mongodbatlas/fw_resource_mongodbatlas_alert_configuration.go

```diff
@@ -555,101 +555,100 @@ func (r *AlertConfigurationRS) ImportState(ctx context.Context, req resource.Imp
-			WebhookURL:               value.WebhookURL.ValueString(),
-		}
 	}
 
+	for i := range tfNotificationSlice {
+		notification := admin.AlertsNotificationRootForGroup{
+			ApiToken:                 tfNotificationSlice[i].APIToken.ValueStringPointer(),
+			ChannelName:              tfNotificationSlice[i].ChannelName.ValueStringPointer(),
+			DatadogApiKey:            tfNotificationSlice[i].DatadogAPIKey.ValueStringPointer(),
+			DatadogRegion:            tfNotificationSlice[i].DatadogRegion.ValueStringPointer(),
+			DelayMin:                 pointy.Int(int(tfNotificationSlice[i].DelayMin.ValueInt64())),
+			EmailAddress:             tfNotificationSlice[i].EmailAddress.ValueStringPointer(),
+			EmailEnabled:             tfNotificationSlice[i].EmailEnabled.ValueBoolPointer(),
+			IntervalMin:              pointy.Int(int(*tfNotificationSlice[i].IntervalMin.ValueInt64Pointer())),
+			MobileNumber:             tfNotificationSlice[i].MobileNumber.ValueStringPointer(),
+			OpsGenieApiKey:           tfNotificationSlice[i].OpsGenieAPIKey.ValueStringPointer(),
+			OpsGenieRegion:           tfNotificationSlice[i].OpsGenieRegion.ValueStringPointer(),
+			ServiceKey:               tfNotificationSlice[i].ServiceKey.ValueStringPointer(),
+			SmsEnabled:               tfNotificationSlice[i].SMSEnabled.ValueBoolPointer(),
+			TeamId:                   tfNotificationSlice[i].TeamID.ValueStringPointer(),
+			TypeName:                 tfNotificationSlice[i].TypeName.ValueStringPointer(),
+			Username:                 tfNotificationSlice[i].Username.ValueStringPointer(),
+			VictorOpsApiKey:          tfNotificationSlice[i].VictorOpsAPIKey.ValueStringPointer(),
+			VictorOpsRoutingKey:      tfNotificationSlice[i].VictorOpsRoutingKey.ValueStringPointer(),
+			Roles:                    tfNotificationSlice[i].Roles,
+			MicrosoftTeamsWebhookUrl: tfNotificationSlice[i].MicrosoftTeamsWebhookURL.ValueStringPointer(),
+			WebhookSecret:            tfNotificationSlice[i].WebhookSecret.ValueStringPointer(),
+			WebhookUrl:               tfNotificationSlice[i].WebhookURL.ValueStringPointer(),
+		}
+		if !tfNotificationSlice[i].NotifierID.IsUnknown() {
+			notification.NotifierId = tfNotificationSlice[i].NotifierID.ValueStringPointer()
+		}
+		notifications = append(notifications, notification)
+	}
 	return notifications, nil
 }
 
-func newThreshold(tfThresholdConfigSlice []tfThresholdConfigModel) *matlas.Threshold {
+func newThreshold(tfThresholdConfigSlice []tfThresholdConfigModel) *admin.GreaterThanRawThreshold {
 	if len(tfThresholdConfigSlice) < 1 {
 		return nil
 	}
 
 	v := tfThresholdConfigSlice[0]
-	return &matlas.Threshold{
-		Operator:  v.Operator.ValueString(),
-		Units:     v.Units.ValueString(),
-		Threshold: v.Threshold.ValueFloat64(),
+	return &admin.GreaterThanRawThreshold{
+		Operator:  v.Operator.ValueStringPointer(),
+		Units:     v.Units.ValueStringPointer(),
+		Threshold: pointy.Int(int(v.Threshold.ValueFloat64())),
```

Comment:
old sdk was receiving a float value, now we are converting to int losing decimal values. We can check if Atlas API does in fact receive float values in some cases or not.

Reply:
For this specific case(`GreaterThanRawThreshold`) Atlas API always expects an integer value

Reply:
👍 . For now I believe it is okay to leave the float type in our terraform model.